File: LeetCode/go/LC2275_Largest_Combination_with_Bitwise_AND_Greater_than_Zero/largest_combination_with_bitwise_and_greater_than_zero.go
package largest_combination_with_bitwise_and_greater_than_zero

import "slices"

func largestCombination(candidates []int) int {
	cnt := [24]int{0}
	for _, v := range candidates {
		for i := 0; v > 0; i++ {
			cnt[i] += v & 1
			v >>= 1
		}
	}
	return slices.Max(cnt[:])
}

File: LeetCode/java/LC2275_Largest_Combination_With_Bitwise_A_N_D_Greater_Than_Zero/LargestCombinationWithBitwiseANDGreaterThanZero.java
package LC2275_LargestCombinationWithBitwiseANDGreaterThanZero;

import java.util.Arrays;

/**
 * <a href="https://leetcode.cn/problems/largest-combination-with-bitwise-and-greater-than-zero/">Largest Combination With Bitwise AND Greater Than Zero</a>
 * 位运算；数组；哈希表；计数
 */
public class LargestCombinationWithBitwiseANDGreaterThanZero {
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.largestCombination(new int[]{16, 17, 71, 62, 12, 24, 14}));
    }
}

class Solution {
    public int largestCombination(int[] candidates) {
        int[] cnt = new int[24];
        for (int c : candidates) {
            for (int i = 0; c > 0; i++) {
                cnt[i] += c & 1;
                c >>= 1;
            }
        }
        return Arrays.stream(cnt).max().getAsInt();
    }
}